policy/policy.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2021 Security Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package policy
+
+import (
+	"errors"
+	"fmt"
+
+	"gopkg.in/yaml.v3"
+
+	"github.com/ossf/scorecard/v2/checks"
+	sce "github.com/ossf/scorecard/v2/errors"
+)
+
+var (
+	errInvalidVersion = errors.New("invalid version")
+	errInvalidCheck   = errors.New("invalid check name")
+	errInvalidScore   = errors.New("invalid score")
+	errInvalidMode    = errors.New("invalid mode")
+	errRepeatingCheck = errors.New("check has multiple definitions")
+)
+
+var modes = map[string]bool{"enforced": true, "disabled": true, "logging": true}
+
+// CheckPolicy defines the policy for a check.
+//nolint:govet
+type CheckPolicy struct {
```

Comment:
Let's define both `CheckPolicy` and `ScorecardPolicy` as proto bufs. We plan to use this data structures across various platforms (browsers, GitHub actions etc.), so let's keep the definition platform agnostic.

Reply:
will do. Note that various platforms could take a file as input; files are platform agnostic. I think protobuf is better too as it's easier to transport as a sequence of bytes  + files lead to all kinds of vulnerabilities.
@oliverchang in case you have something to comment/warn about

Reply:
Using protobufs internally seems fine, as long as the actual format exposed to end users is JSON. Protobufs can be serialized to JSON, and I believe there is a stable ordering of properties (based on the field numbers).

Reply:
If I'm not mistaken @azeemsgoogle 's comment is using the protobuf for the exported structures `CheckPolicy` and `ScorecardPolicy`. Can you clarify?

Reply:
To be clear, my reasoning to define these as protobufs is to guarantee that the data structure is consistently maintained across platforms. The intention is *not* to serialize these as protobufs. The best format to serialize will depend on the usecase, and in most cases we will probably stick to JSON. I believe this is what @oliverchang is also trying to say.

Reply:
I don't understand, probably better to take this offline over GVC. I thought the point of using  protobuf was that we can easily have clients use javascript or other languages to read it from a serialized string.
If it stays internal, isn't the limiting factor golang, ie everyone needs to use our golang APIs anyway...

Reply:
Talked to @azeemsgoogle offline. I'm fine with this. Thanks Azeem!